### PR TITLE
Improve caching of test data

### PR DESCRIPTION
**Pull request by maxulysse**
simplify caching thanks to @grst and @sateeshperi

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,7 +40,6 @@ jobs:
         id: hash_workspace
         run: |
           echo "digest=$(echo sarek3_${{ github.workspace }} | md5sum | cut -c 1-25)" >> $GITHUB_OUTPUT
-
       - name: Cache test data
```
Comment (maxulysse) [inline .github/workflows/ci.yml:43]:
```suggestion

      - name: Cache test data
```

**Review by FriederikeHanssen — APPROVED** [review]
looks good, can we merge this?

**Comment by asp8200** [comment]
Not sure what this was about, so I search around a bit. From [this](https://github.com/actions/checkout#usage), I got the impression that `fetch-depth` is set to `1` by default. I couldn't find a definition of the `fetch-depth` equal to `1` means 🤔 Be that as it may, I see no harm in setting it explicitly.

**Comment by maxulysse** [comment]
Yeah not sure about this one. Let's keep it aside for now

**Pull request closed without merging**